File: LondonFhirService.Core/Models/Foundations/Audits/Audit.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;
using LondonFhirService.Core.Models.Bases;

namespace LondonFhirService.Core.Models.Foundations.Audits
{
    public class Audit : IKey, IAudit
    {
        public Guid Id { get; set; }
        public string CorrelationId { get; set; }
        public string AuditType { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public string Message { get; set; } = string.Empty;
        public string FileName { get; set; }
        public string LogLevel { get; set; } = "Information";
        public string CreatedBy { get; set; } = string.Empty;
        public string UpdatedBy { get; set; } = string.Empty;
        public DateTimeOffset UpdatedDate { get; set; }
        public DateTimeOffset CreatedDate { get; set; }
    }
}

File: LondonFhirService.Manage.Client/src/models/foundations/audits/Audit.ts
// Foundation model - mirrors LondonFhirService.Core.Models.Foundations.Audits.Audit as the
// /api/audits endpoint serialises it. CorrelationId and FileName have no server side default, so
// they can come back null.
export type Audit = {
    id: string;
    correlationId: string | null;
    auditType: string;
    title: string;
    message: string;
    fileName: string | null;
    logLevel: string;
    createdBy: string;
    createdDate: string;
    updatedBy: string;
    updatedDate: string;
};
